app.py

```diff
@@ -1,134 +1,172 @@
-import os
-from flask import Flask, render_template, jsonify, request
+from flask import Flask, render_template, jsonify, request, Response
 import requests
-from urllib.parse import urlparse, quote_plus
+from urllib.parse import quote_plus
 import logging
 from flask_caching import Cache
 import time
 
-logging.basicConfig(level=logging.DEBUG, format='%(asctime)s - %(levelname)s - %(message)s')
+logging.basicConfig(
+    level=logging.DEBUG, format="%(asctime)s - %(levelname)s - %(message)s"
+)
 
 app = Flask(__name__)
 
 # Configure caching
-cache = Cache(app, config={'CACHE_TYPE': 'SimpleCache'})
+cache = Cache(app, config={"CACHE_TYPE": "SimpleCache"})
 
-@app.route('/')
-def index():
-    return render_template('index.html')
 
-@app.route('/landmarks')
-def get_landmarks():
+@app.route("/")
+def index() -> str:
+    return render_template("index.html")
+
+
+@app.route("/landmarks")
+def get_landmarks() -> Response:
     # Get coordinates from query parameters
-    lat = float(request.args.get('lat'))
-    lon = float(request.args.get('lon'))
-    radius = int(float(request.args.get('radius', 10000)))  # Default radius: 10km, converted to integer
-    filters = request.args.get('filters', '').split(',')
+    lat: float = float(request.args.get("lat"))
+    lon: float = float(request.args.get("lon"))
+    radius: int = int(
+        float(request.args.get("radius", 10000))
+    )  # Default radius: 10km, converted to integer
+    filters: list[str] = request.args.get("filters", "").split(",")
 
```

Comment:
Great work on adding type annotations and improving logging! This improves code readability and maintainability.

However, the pull request is intended to implement a new 'route to landmark' feature. You should ensure that the backend logic for calculating routes is implemented or detailed next, as currently, this file does not seem to fully support that feature.